dask/base.py

```diff
@@ -53,6 +54,43 @@
     from _typeshed import ReadableBuffer
 
 
+class PrintPostCompute(Callback):
+    """Callback that prints out a result after it's computed.
+    To enqueue a callback, use dask.base.print_on_compute(...)."""
+
+    queue: dict[Hashable, Any] = {}
+
+    def _posttask(self, key, value, _dsk, _state, _id):
+        printers = PrintPostCompute.queue.pop(key, [])
+        for format_string in printers:
+            print(format_string % value)
+
+    def _finish(self, _dsk, _state, _errored):
+        PrintPostCompute.queue.clear()
+
+
+def print_on_compute(format_string="", dask_collection: DaskCollection | None = None):
+    """Enqueue a print statement after this collection result if computed.
+
+    Parameters
+    ----------
+    format_string : str, default: "%s"
+        String to use when formatting the object.
+
+    dask_collection : DaskCollection
+        Any lazy Dask object.
+    """
+    if not format_string:
+        format_string = "%s"
+    if dask_collection is None:
+        return
+    key: Hashable = dask_collection.__dask_keys__()[0]
+    if key in PrintPostCompute.queue:
+        PrintPostCompute.queue[key].append(format_string)
+    else:
+        PrintPostCompute.queue[key] = [format_string]
```

Comment:
This will exclusively work for dask.delayed, dask.dataframe with a single partition, and dask.array with a single chunk and either 0 or 1 dimensions.
In general you should use
```python
from dask.core import flatten
for key in flatten(dask_collection.__dask_keys__()):
    ...
```

Reply:
Thanks. 👍